pkg/s3/s3client.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company and Gardener Project Authors. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+type S3Client struct {
+	S3 *s3.S3
+}
+
+func NewS3Client(accessKey, secret, endpoint, region string) (*S3Client, error) {
+	noSSL := false
+
+	pair := strings.Split(endpoint, "://")
+
+	if len(pair) > 1 {
+		noSSL = (pair[0] == "http")
+	}
+
+	pathStyle := true
+	token := ""
+	creds := credentials.NewStaticCredentials(accessKey, secret, token)
+	sess, err := session.NewSessionWithOptions(session.Options{
+		Config: aws.Config{
+			Region:           &region,
+			Credentials:      creds,
+			Endpoint:         &endpoint,
+			DisableSSL:       &noSSL,
+			S3ForcePathStyle: &pathStyle,
+		},
+		// Profile: "profile_name",
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("Unable to create S3 session instance: %v", err)
```

Comment:
The style of wrapping errors used elsewhere in this extension is by using the `github.com/pkg/errors` package.
```suggestion
		return nil, errors.Wrap(err, "could not create S3 session")
```